src/uct/sm/knem/knem_ep.c

```diff
@@ -86,13 +98,17 @@ ucs_status_t uct_knem_ep_put_zcopy(uct_ep_h tl_ep, const uct_iov_t *iov, size_t
                                    uct_completion_t *comp)
 {
     uct_knem_key_t *key = (uct_knem_key_t *)rkey;
+    uct_base_iface_t *iface;
     ucs_status_t status;
+    size_t length;
 
-    UCT_CHECK_PARAM(1 == iovcnt, "iov[iovcnt] has to be 1 at this time");
-    size_t length = iov[0].length;
+    iface = ucs_derived_of(tl_ep->iface, uct_base_iface_t);
 
-    status = uct_knem_rma(tl_ep, iov[0].buffer, length, remote_addr, key, 1);
-    UCT_TL_EP_STAT_OP_IF_SUCCESS(status, ucs_derived_of(tl_ep, uct_base_ep_t), PUT, ZCOPY, length);
+    UCT_CHECK_IOV_SIZE(iovcnt, iface->config.max_iov, "uct_knem_ep_put_zcopy");
+
+    status = uct_knem_rma(tl_ep, iov, iovcnt, &length, remote_addr, key, 1);
```

Comment:
@yosefe As I see the `length` is used in release mode by `UCT_KNEM_ZERO_LENGTH_POST(length)`
This macro returns `UCS_OK` in case of the data total zero length.
So, Shuld I use `uct_iov_total_length(iov)` anyway to calculate `length` at caller functions?